redash/query_runner/python.py

```diff
@@ -290,6 +289,25 @@ def get_current_user(self):
     def test_connection(self):
         pass
 
+    def validate_result(self, result):
+        """Validate the result after executing the query.
+
+        Parameters:
+        :result dict: The result dict.
+        """
+        if not result:
+            raise Exception("local variable `result` should not be empty.")
```

Comment:
This line will never evaluate `True` because `result` is set explicitly by default on line 107.

Reply:
Sorry for the really late response. It seems the issue still exists.
* IMO, the validation is a MSUT for the "user generated input" to improve the robustness of the application.
* Indeed the expected return variable `result` has been set to a default value "before" the python execution. But anything can happen "after" the run, like setting `result` to `None` or `list`.
* I myself did encounter a few times that reuse this common word "result" in other logic by accident.